actionpack/test/dispatch/cookies_test.rb

```diff
@@ -810,6 +810,24 @@ def test_encrypted_cookie_using_hybrid_serializer_can_read_from_json_dumped_valu
     assert_nil @response.cookies["foo"]
   end
 
+  def test_encrypted_cookie_using_json_serializer_will_drop_marshal_dumped_value
+    @request.env["action_dispatch.cookies_serializer"] = :json
+
+    key_generator = @request.env["action_dispatch.key_generator"]
+    secret = key_generator.generate_key(@request.env["action_dispatch.authenticated_encrypted_cookie_salt"], 32)
+
+    encryptor = ActiveSupport::MessageEncryptor.new(secret, cipher: "aes-256-gcm", serializer: Marshal)
+    marshal_value = encryptor.encrypt_and_sign("bar")
+    @request.headers["Cookie"] = "foo=#{::Rack::Utils.escape marshal_value}"
```

Comment:
I wonder if this is overkill for this test. All we really need to test is that invalid JSON returns a nil cookie?

Reply:
Hmm, you mean just feed the test a bad json rather than a cookie? I'm not sure how to do that, it'll trigger a different error earlier in the flow.
I tried `@request.headers["Cookie"] = "foo=<bad_value>"` but this doesn't fail without my test (so probable raises one of `ActiveSupport::MessageEncryptor::InvalidMessage`, ActiveSupport::MessageVerifier::InvalidSignature`)
I'm not sure how I can make this test lighter.
Any suggestion?

PS: I also felt like testing the exact case that brought this problem up 🤷

Reply:
Ah, you're totally right. It has to make it through the `MessageDecryptor` first. I think this test is good then.